### PR TITLE
Deprecation of some EKF2_AID_MASK bits

### PR DESCRIPTION
**Pull request by hamishwillee**
`EKF2_AID_MASK` largely deprecated - this tries to fix the cases for which that is true. Open questions below about what replaces "rotate" - i.e. more work needed here.

Follows on from #2456

- [x] Ready to merge when this goes in: https://github.com/PX4/PX4-Autopilot/pull/21514

### CODE REVIEW

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 1]
```diff
@@ -433,7 +433,7 @@ A good tuning is obtained as follows:
 
 Position, velocity or orientation measurements from an external vision system, e.g. Vicon, can be used:
 
-* External vision system horizontal position data will be used if bit position 3 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system horizontal position data will be used if bit position 0 in the [EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) parameter is true.
 * External vision system vertical position data will be used if the [EKF2_HGT_REF](../advanced_config/parameter_reference.md#EKF2_HGT_REF) parameter is set to 3.
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md:437]: Should `EKF2_EV_CTRL` bit 1 also be set as well as `EKF2_HGT_REF`?

File: en/computer_vision/visual_inertial_odometry.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ The following parameters must be set to use external position information with E
 
 Parameter | Setting for External Position Estimation
 --- | ---
-[EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) | Set *vision position fusion*, *vision velocity fusion*, *vision yaw fusion* and *external vision rotation* according to your desired fusion model.
+[EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) | Set *horizontal position fusion*, *vertical vision fusion*, *velocity fusion*, *yaw fusion* and *external vision rotation* according to your desired fusion model.
```
Comment (hamishwillee) [inline en/computer_vision/visual_inertial_odometry.md:93]: Same as above - how do I fix up the last point "external vision rotation".

File: en/ros/external_position_estimation.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ The following parameters must be set to use external position information with E
 
 Parameter | Setting for External Position Estimation
 --- | ---
-[EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) | Set *vision position fusion*, *vision velocity fusion*, *vision yaw fusion* and *external vision rotation* according to your desired fusion model.
+[EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) | Set *horizontal position fusion*, *vertical vision fusion*, *velocity fusion*, *yaw fusion* and *external vision rotation* according to your desired fusion model.
```
Comment (hamishwillee) [inline en/ros/external_position_estimation.md:82]: Same problem as above - what's the story with rotation: https://github.com/PX4/PX4-user_guide/pull/2459/files#r1170733088

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 1 above)
Comment (bresch) [reply en/advanced_config/tuning_the_ecl_ekf.md:437]: Yes, this has changed as well. To activate EV height fusion, `EKF2_EV_CTRL` bit 1 needs to be set. It is not required to set `EKF2_HGT_REF` to external vision, any reference can be used (e.g.: baro, GNSS height).

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 4]
```diff
@@ -433,24 +433,24 @@ A good tuning is obtained as follows:
...
-* External vision system velocity data will be used if bit position 8 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
-* External vision system orientation data will be used for yaw estimation if bit position 4 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
-* External vision reference frame offset will be estimated and used to rotate the external vision system data if bit position 6 in the [EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) parameter is true.
+* External vision system velocity data will be used if bit position 2 in the [EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) parameter is true.
+* External vision system orientation data will be used for yaw estimation if bit position 3 in the [EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) parameter is true.
+* External vision reference frame offset will be estimated and used to rotate the external vision system data if bit position 3 in the [EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) parameter is true.
```
[the remark this answers is not in the document]
Comment (bresch) [reply en/advanced_config/tuning_the_ecl_ekf.md:441]: `EV_ROTATE` doesn't exist anymore. IIRC we just assume that the data is aligned wrt North if `EV_YAW` is not set. Maybe @dagar can confirm that?

File: en/computer_vision/visual_inertial_odometry.md (hunk 2 above)
Comment (bresch) [reply en/computer_vision/visual_inertial_odometry.md:93]: remove "external vision rotation"

Comment (hamishwillee) [inline en/computer_vision/visual_inertial_odometry.md:93]:
```suggestion
[EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) | Set *horizontal position fusion*, *vertical vision fusion*, *velocity fusion*, and *yaw fusion* according to your desired fusion model.
```

Comment (hamishwillee) [reply en/computer_vision/visual_inertial_odometry.md:93]:
```suggestion
[EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) | Set *horizontal position fusion*, *vertical vision fusion*, *velocity fusion*, and *yaw fusion* according to your desired fusion model.
```

File: en/ros/external_position_estimation.md (hunk 3 above)
Comment (hamishwillee) [reply en/ros/external_position_estimation.md:82]:
```suggestion
[EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) | Set *horizontal position fusion*, *vertical vision fusion*, *velocity fusion*, and *yaw fusion*, according to your desired fusion model.
```

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 5]
```diff
@@ -431,30 +431,17 @@ A good tuning is obtained as follows:
 
 ### External Vision System
```
Comment (hamishwillee) [inline en/advanced_config/tuning_the_ecl_ekf.md:432]: THanks @bresch - much appreciated. I have rewritten this section. Are you happy to merge this PR?

File: en/advanced_config/tuning_the_ecl_ekf.md [hunk 6]
```diff
@@ -431,30 +431,17 @@ A good tuning is obtained as follows:
...
-:::
+The measurements that are fused are configured by setting the appropriate bits of [EKF2_EV_CTRL](../advanced_config/parameter_reference.md#EKF2_EV_CTRL) to `true`:
+
+- `0`: Horizontal position data 
+- `1`: Vertical position data.
+  Height sources may additionally be configured using [EKF2_HGT_REF](../advanced_config/parameter_reference.md#EKF2_HGT_REF) (see section [Height](#height)).
+- `2`: Velocity data 
+- `3`: Yaw data
+
+Note that if yaw data is used (bit 3) the heading is with respect to the external vision frame; otherwise the heading is relative to North.
```
Comment (beniaminopozzan) [inline en/advanced_config/tuning_the_ecl_ekf.md:444]: @hamishwillee as a future note, right now it should be possible to transit from indoor (EV) to outdoor (GPS) and vice versa i.e. have both GPS an EV potentially enabled at the same time, which I think it should be mentioned in the documentation.
I'm still trying to figure out how it works though.

Comment (hamishwillee) [reply en/advanced_config/tuning_the_ecl_ekf.md:444]: That's cool, but not until we know how. Can I leave that with you for now?

**Comment by mrpollo** [comment]
Relevant
* https://github.com/PX4/PX4-Autopilot/pull/20497
* https://github.com/PX4/PX4-Autopilot/pull/20501
* https://github.com/PX4/PX4-Autopilot/pull/21032

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 6 above)
Comment (bresch) [reply en/advanced_config/tuning_the_ecl_ekf.md:444]: I would suggest that we don't mention it in the docs until we're really sure that this works as expected.

**Review by bresch — APPROVED** [review]

File: en/advanced_config/tuning_the_ecl_ekf.md (hunk 5 above)
Comment (bresch) [reply en/advanced_config/tuning_the_ecl_ekf.md:432]: Yes, looks good now.

**Comment by bresch** [comment]
Ok, we decided to deprecate the rest before the release to have something more consistent: https://github.com/PX4/PX4-Autopilot/pull/21514
I'll update this branch

**Comment by mrpollo** [comment]
Please merge once https://github.com/PX4/PX4-Autopilot/pull/21514 has been merged

**Comment by hamishwillee** [comment]
Awesome. Thanks for reviews. Happy for anyone to merge this once that associated PR goes in.

**Comment by hamishwillee** [comment]
FYI, merged as I see the accompanying PR has gone in. Thanks.

**Comment by mrpollo** [comment]
Thank you @hamishwillee 🌮

**Comment by hamishwillee** [comment]
You're welcome - all kudos to @bresch really.

**Pull request merged**